src/exe/main.cc

```diff
@@ -381,12 +381,24 @@ static int start_profiler_internal(DDProfContext *ctx, bool &is_profiler) {
           [&reply](const ddprof::RequestMessage &) { return reply; });
     } catch (const ddprof::DDException &e) {
       if (in_wrapper_mode) {
-        // Failture in wrapper mode is not fatal:
-        // LD_PRELOAD may fail because target exe is statically linked
-        // (eg. go binaries)
-        LG_WRN("Unable to connect to profiler library (target executable might "
-               "be statically linked and library cannot be preloaded). "
-               "Allocation profiling will be disabled.");
+        if (!process_is_alive(ctx->params.pid)) {
+          // Tell the user that process died
+          // Most of the time this is an invalid command line
+          LG_ERR(
+              "Target process(%d) is not alive. Check command line validity.",
+              ctx->params.pid);
+          // Avoiding returning
```

Comment:
I'm not sure if we should stop here.

Reply:
99% of the time, the user just did a wrong command line.

Reply:
@nsavoire do you think we should return -1 in this case ?

Reply:
I think the first thing to fix is to print an error message when `execvp` fails.
Currently logs [emitted](https://github.com/DataDog/ddprof/blob/main/src/exe/main.cc#L467) in that case are not printed, probably because logger is not configured at this point...

Reply:
In the case where process is dead and library connection failed, I would not print anything and let the execution continue.

Reply:
If we're here, then `pid` is set to the value of `getpid()` in the parent, so either
1. `kill()` failed because the pid is invalid (meaning the original process died)
2. `kill()` failed for some permissions-related reason???

I agree that the overwhelming commonality is that the user specified something that couldn't be executed, like a shell builtin or an incorrect command.

I suppose that rarely it may also be the case that the target process was very short-lived and the underlying cgroup didn't admit enough scheduler slots for this child process to emit the check before the parent process died.

Personally, I think we should stop here.  It used to make sense to keep going since we only had CPU profiling, but now I don't think there's such a guarantee, right?  Plus we're at the point now where we have the most information about the state of the failure.

On the other hand
1. we should emit a better error message at time of `execve()` failure in the parent
2. if the main application _was_ short-lived and it mattered, hopefully the user emitted their own errors

so maybe it's more useful to say more elsewhere rather than here.

Reply:
On the small chance that there is a short lived process that forked, I will continue with a warning.
Thanks for the feedback! I'm also adding a message to exec failures.